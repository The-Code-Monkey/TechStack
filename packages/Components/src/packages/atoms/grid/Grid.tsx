import React, { useMemo } from 'react';
import { GridProps } from '@aw-web-design/styled-system';

import { BoxProps } from '../../primatives';

import { StyledGrid } from './styled';
import { getTemplate } from './utils';

export interface Props extends BoxProps, GridProps {
  columns: GridProps['gridTemplateColumns'];
  rows?: GridProps['gridTemplateRows'];
  areas?: GridProps['gridTemplateAreas'];
}

const Grid = ({ children, autoid, columns, rows, areas, ...rest }: Props) => {
  const getTemplateColumnsMemo = useMemo(() => getTemplate(columns), [columns]);
  const getTemplateRowsMemo = useMemo(() => getTemplate(rows), [rows]);

  return (
    <StyledGrid
      autoid={`${autoid}_table`}
      gridTemplateColumns={getTemplateColumnsMemo}
      gridTemplateRows={getTemplateRowsMemo}
      gridTemplateAreas={areas}
      display="grid"
      {...rest}
    >
      {children}
    </StyledGrid>
  );
};

export default Grid;
