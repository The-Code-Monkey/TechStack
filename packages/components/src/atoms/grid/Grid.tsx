import { GridProps } from '@bunstack/styled-system';
import { useMemo } from 'react';

import type { BoxProps } from '../../primal';

import { StyledGrid } from './styled';
import { getTemplate } from './utils';
import React from 'react';

export interface Props extends BoxProps, GridProps {
  columns: GridProps['gridTemplateColumns'];
  rows?: GridProps['gridTemplateRows'];
  areas?: GridProps['gridTemplateAreas'];
}

const Grid = ({ children, testid, columns, rows, areas, ...rest }: Props) => {
  const getTemplateColumnsMemo = useMemo(() => getTemplate(columns), [columns]);
  const getTemplateRowsMemo = useMemo(() => getTemplate(rows), [rows]);

  return (
    <StyledGrid
      testid={`${testid}_table`}
      gridTemplateColumns={getTemplateColumnsMemo}
      gridTemplateRows={getTemplateRowsMemo}
      gridTemplateAreas={areas}
      display='grid'
      {...rest}
    >
      {children}
    </StyledGrid>
  );
};

export default Grid;
