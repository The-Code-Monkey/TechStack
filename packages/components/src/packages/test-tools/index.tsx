import { mount, shallow, ShallowWrapper, ReactWrapper } from 'enzyme';
import React, { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../theme';
import { ConfigContext } from '../utils';

const config = {
  outputDir: '/src/packages/',
  iconDir: 'feather',
};

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: any) =>
      val &&
      typeof val === 'object' &&
      'props' in val &&
      Object.keys(val.props).some((prop) => keys.some((key) => key === prop)),
    print: (val: any, serialize: any) => {
      keys.forEach((key) => {
        delete val.props[key];
      });
      return serialize(val);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors?.common,
    modes: theme.colors?.modes,
    ...theme.colors?.modes.light,
  },
};

export const shallowWithTheme: (children: ReactNode) => ShallowWrapper = (
  children: ReactNode
) =>
  shallow(<ThemeProvider theme={formattedTheme}>{children}</ThemeProvider>)
    .dive()
    .shallow();

export const mountWithTheme: (children: ReactNode) => ReactWrapper = (
  children: ReactNode
) => {
  expect.addSnapshotSerializer(removeProperties());

  return mount(
    <ConfigContext.Provider value={config}>
      <ThemeProvider theme={formattedTheme}>
        <>{children}</>
      </ThemeProvider>
    </ConfigContext.Provider>
  );
};
