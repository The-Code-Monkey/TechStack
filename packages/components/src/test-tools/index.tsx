import isPropValid from '@emotion/is-prop-valid';
import { render, RenderResult } from '@testing-library/react';
import { ReactNode } from 'react';
import { StyleSheetManager, ThemeProvider } from 'styled-components';
import { expect } from 'bun:test';

import theme from '../theme';
import { ConfigContext } from '../utils';

const config = {
  outputDir: '/src/packages/',
  iconDir: 'feather',
};

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: { props: unknown }) =>
      val &&
      typeof val === 'object' &&
      'props' in val &&
      Object.keys(val.props).some(prop => keys.some(key => key === prop)),
    serialize: (val, config, indentation, depth, refs, printer) => {
      keys.forEach(key => {
        delete val.props[key];
      });
      return printer(val, config, indentation, depth, refs);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors?.common,
    modes: theme.colors?.modes,
    ...theme.colors?.modes.light,
  },
};

export const mountWithTheme = (children: ReactNode): RenderResult => {
  // expect.addSnapshotSerializer(removeProperties());

  return render(
    <StyleSheetManager shouldForwardProp={isPropValid}>
      <ConfigContext.Provider value={config}>
        <ThemeProvider theme={formattedTheme}>
          <>{children}</>
        </ThemeProvider>
      </ConfigContext.Provider>
    </StyleSheetManager>
  );
};
