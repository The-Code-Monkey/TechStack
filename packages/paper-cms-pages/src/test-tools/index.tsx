import { ConfigContext } from '@techstack/components';
import { render } from '@testing-library/react';
import { ReactNode } from 'react';

import theme from '../theme';
import { ThemeProvider } from '../workarounds/styled-components';

const config = {
  outputDir: '/src/packages/',
  iconDir: 'feather',
};

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: { props: unknown }) =>
      val &&
      typeof val === 'object' &&
      'props' in val &&
      Object.keys(val.props).some(prop => keys.some(key => key === prop)),
    serialize: (val, config, indentation, depth, refs, printer) => {
      keys.forEach(key => {
        delete val.props[key];
      });
      return printer(val, config, indentation, depth, refs);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors?.common,
    modes: theme.colors?.modes,
    ...theme.colors?.modes.light,
  },
};

export const mountWithTheme = (children: ReactNode) => {
  expect.addSnapshotSerializer(removeProperties());

  return render(
    <ConfigContext.Provider value={config}>
      <ThemeProvider theme={formattedTheme}>
        <>{children}</>
      </ThemeProvider>
    </ConfigContext.Provider>
  );
};
