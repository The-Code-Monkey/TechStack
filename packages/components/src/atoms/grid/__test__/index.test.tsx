import { mountWithTheme } from '../../../test-tools';
import Grid, { Cell } from '../index';

describe('<Grid />', () => {
  it('renders correctly', () => {
    const { asFragment } = mountWithTheme(
      <Grid columns={['auto', '1fr 1fr', '1fr 1fr 2fr']}>
        <Cell>Cell 1</Cell>
        <Cell>Cell 2</Cell>
        <Cell>Cell 3</Cell>
        <Cell>Cell 4</Cell>
        <Cell>Cell 5</Cell>
      </Grid>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
