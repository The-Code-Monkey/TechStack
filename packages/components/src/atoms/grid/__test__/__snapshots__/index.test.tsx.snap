// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe brroDR sc-eDvSVe eytpQn"
    data-testid="undefined_table"
  >
    <div
      class="sc-bcXHqe iXvHqq sc-jSUZER hiKBRA"
      data-testid="cell_1_cell"
    >
      Cell 1
    </div>
    <div
      class="sc-bcXHqe iXvHqq sc-jSUZER hiKBRA"
      data-testid="cell_2_cell"
    >
      Cell 2
    </div>
    <div
      class="sc-bcXHqe iXvHqq sc-jSUZER hiKBRA"
      data-testid="cell_3_cell"
    >
      Cell 3
    </div>
    <div
      class="sc-bcXHqe iXvHqq sc-jSUZER hiKBRA"
      data-testid="cell_4_cell"
    >
      Cell 4
    </div>
    <div
      class="sc-bcXHqe iXvHqq sc-jSUZER hiKBRA"
      data-testid="cell_5_cell"
    >
      Cell 5
    </div>
  </div>
</DocumentFragment>
`;
